### PR TITLE
use varint to represent data version in the key

### PR DESCRIPTION
**Pull request by zlcook**
As title, Refer to Issue #485

### CODE REVIEW

**Comment by zlcook** [comment]
jenkins go!

File: src/storage/test/AdHocSchemaManager.cpp [hunk 1]
```diff
@@ -9,6 +9,22 @@
...
+}
+
+bool AdHocSchemaManager::isSupportTimeSeries(GraphSpaceID space) {
+    folly::RWSpinLock::ReadHolder rh(seriesLock_);
+    auto it = timeSeries_.find(space);
+    if (it == timeSeries_.end()) {
+        // Not found
+        return false;
+    } else {
+        return timeSeries_[space];
```
Comment (wadeliuyi) [inline src/storage/test/AdHocSchemaManager.cpp]: maybe you can return iter->second directly

Comment (zlcook) [reply src/storage/test/AdHocSchemaManager.cpp]: Good catch :)

File: src/common/base/NebulaKeyUtils.inl [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+namespace nebula {
+
+template<typename T>
+typename std::enable_if<std::is_integral<T>::value, int32_t>::type
+NebulaKeyUtils::encodeVersion(T v, uint8_t *encode) {
+    size_t len = folly::encodeVarint(v, encode);
+    DCHECK_GT(len, 0UL);
+    uint8_t temp;
+    // Keep the original order
+    for (size_t i = 0; i < len / 2; i++) {
```
Comment (critical27) [inline src/common/base/NebulaKeyUtils.inl]: Why do we need to reverse the order both in encode and decode?

Comment (zlcook) [reply src/common/base/NebulaKeyUtils.inl]: > Why do we need to reverse the order both in encode and decode?

Sorry for the late reply! 
The reason：
* 300 is greater than 200, however, the numbers(ps: 200, 300) encoded by `folly::encodeVarint` does not guarantee the original order.  In order to maintain the original order, reverse the byte order.  [folly::encodeVarint principle](https://developers.google.com/protocol-buffers/docs/encoding?csw=1#varints)

* The `key` in RocksDb sorted in lexicographic order.

File: src/common/base/NebulaKeyUtils.inl [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        encode[len-i-1] = temp;
+    }
+    return len;
+}
+
+template<typename T>
+typename std::enable_if<std::is_integral<T>::value>::type
+NebulaKeyUtils::decodeVersion(const char* data, int32_t len, T& v) {
+    const uint8_t* encode = reinterpret_cast<const uint8_t*>(data);
+    uint8_t copy[10];
```
Comment (wadeliuyi) [inline src/common/base/NebulaKeyUtils.inl]: Is it sure that len is less than 10?

Comment (zlcook) [reply src/common/base/NebulaKeyUtils.inl]: Yes, the encoded value of the 8-byte largest integer occupies up to 10 bytes.

**Comment by zlcook** [comment]
Jenkins go

**Comment by dangleptr** [comment]
Close it.  We don't change the format of our key in 1.x any more.  Thanks~

**Pull request closed without merging**